### DISCUSSION TITLE
offline maps

### DISCUSSION THREAD

**Discussion by Cer-o** [devemux86/cruiser#194] [Q&A] [answered]
Hello everyone, I want to resume my contact with Cruiser and I see a lot of news since the last time.
My first question is related to offline maps.
I see that there are several options but I don't know the differences between them. I don't ask for anything special, just that you respect the route I have programmed (mainly I like curvy roads) and that it be easy to use.
Here is a screenshot of the map options that appear to me. Can you explain to me what each of them are and their main differences?
Thank you so much.

![image](https://github.com/devemux86/cruiser/assets/122309655/d285b3f0-6f82-440a-95d6-3e0e31980bfd)

**Comment by devemux86** [comment, marked as answer]
@Cer-o please ask anything you like to know!

Cruiser's focus is to work offline and give users many options so they can customize the app to their needs.

You only need 3 simple steps: Download offline maps + Offline router + Select profile.

1. **Maps**: Download offline maps ("Mapas" button) for any region in the world.
It also automatically downloads POIs for these maps, used in **offline search**.

2. **Router**: you will need an offline router while being outside.
Please see the BRouter guide for instructions: https://github.com/devemux86/cruiser/discussions/137

3. **Profile**: select in routing settings "**Motorcycle**" and "**BRouter (offline)**" and you are ready to go!

Note:
With the "Puntos de interés" button you can manage the poi separately.
GraphHopper is an alternative offline router, but BRouter is more powerful with detailed options.

Over time you can explore the other settings and please ask when something is not clear.
If you need anything else please let me know! :slightly_smiling_face: 

---

_Remember that the rerouting in Cruiser is **offline** and does **NOT** change your curvy routes!_

**Comment by Cer-o** [comment]
Thank you so much.
I have downloaded and installed BRouter and downloaded the maps of Europe.
I understand that this option is for when you decide to drive offline, but I can't find the option to activate it.

**Reply by devemux86** [reply]
@Cer-o it is in "Settings | Routing | Service | BRouter (offline)".

**Reply by 0709wiwiwi** [reply]
In the Desktop version "Service" is logically located in.
Routing > Set Service
_To know: 
Only GraphHopper (**offline**) is available in the desktop version._

**Reply by Cer-o** [reply]
> @Cer-o está en "Configuración | Enrutamiento | Servicio | BRouter (offline)".

And how do I choose to navigate online or offline?

**Reply by devemux86** [reply]
@Cer-o 

> And how do I choose to navigate online or offline?

Navigation is always offline. Do you mean the rerouting when you go away from the route?

The rerouting is offline when you select as routing service the "BRouter (offline)".

Cruiser rerouting does not change your curved routes, it only calculates up to the next waypoint.

**Reply by Cer-o** [reply]
> @Cer-o
> 
> > And how do I choose to navigate online or offline?
> 
> Navigation is always offline. Do you mean the rerouting when you go away from the route?
> 
> The rerouting is offline when you select as routing service the "BRouter (offline)".
> 
> Cruiser rerouting does not change your curved routes, it only calculates up to the next waypoint.

No, I mean what should I do to choose to navigate with the online maps or with the downloaded maps.
I remember that in kurviger it was easy to choose and there were differences if you recalculated with online or offline maps,

I don't know if I can explain myself.

**Reply by devemux86** [reply]
@Cer-o when you download offline maps, they are activated automatically with the Biker theme.

You can also manage them with the download manager, via menu "Map | Downloads | Maps".

Also use the Map menu to select a map theme (Biker or MotoRider are excellent) and map layers.

<img src="https://github.com/devemux86/cruiser/assets/3484020/d5271608-1392-4d1d-a46e-a37d0759006d" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/b7a70735-bc69-42c9-97be-b341d6fc1730" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/6f7a82fe-d32b-47f1-bec7-7d6eb76fbdba" width="33%"/>

**Reply by devemux86** [reply]
> there were differences if you recalculated with online or offline maps

The offline maps are only needed to display offline the map on the screen.

The offline route calculation needs to install and select the "BRouter (offline)".

**Reply by SF4User** [reply]
Is this a preview? I don't find in the Menu of Cruiser.
I'm downloaded to day all the Mapsforge offline maps new, but this option was not highlighted.
Remark: the Cruiser Navigation is a heavy App . . . comes day by day perfecter as before 👍💪😉
Thank You for an answer.

**Reply by devemux86** [reply]
@SF4User 

> I don't find in the Menu of Cruiser.

Have you updated to the latest version?
What exactly can't you find?

> Remark: the Cruiser Navigation is a heavy App . . . comes day by day perfecter as before 👍💪😉

Thanks for using Cruiser! :slightly_smiling_face:

**Comment by Cer-o** [comment]
I have already found the way:

You have to download the map and activate it. That way you circulate offline.
If you want to circulate with the online maps, you have to go to Maps / Online Maps / and choose among the many that exist.

I take this opportunity to ask the differences between the downloadable maps (Mapsforge, OpenAndroMaps or Freizeitkarte) and which one is more complete.

Thank you so much.

**Reply by devemux86** [reply]
Offline maps in Cruiser are provided by 3 sources:
- [Mapsforge](https://download.mapsforge.org/) with Biker or MotoRider themes
They are the maps you already know with layers that appear early, like fuel stations.

- [OpenAndroMaps](https://www.openandromaps.org/en/) with Elevate or Voluntary themes
- [Freizeitkarte](https://www.freizeitkarte-osm.de/android/en/index.html) with Freizeitkarte themes

OpenAndroMaps and Freizeitkarte are detailed maps with contour lines and focus on outdoor activities.
They are best used with their own downloaded map themes.

It is recommended to start with the default Mapsforge offline maps. :slightly_smiling_face:

**Comment by Cer-o** [comment]
One more question about BRouter, how do I know if the maps I have downloaded need to be updated?

**Reply by devemux86** [reply]
I have not included a notification for the BRouter downloads, because they are updated **daily**.
Every day there are BRouter updates!

Although it is not absolutely necessary to look for the updates every day. :slightly_smiling_face:

**Reply by devemux86** [reply]
There are notifications for all other downloads, please see the images here:

- https://github.com/devemux86/cruiser/discussions/86#discussioncomment-6023184

**Comment by Cer-o** [comment]
And in each update the map is downloaded in its entirety?

![image](https://github.com/devemux86/cruiser/assets/122309655/067817db-350c-494c-9750-bbfa753c2a85)

**Reply by devemux86** [reply]
> And in each update the map is downloaded in its entirety?

The offline maps are downloaded again in full with the updates.

**Comment by Cer-o** [comment]
Tengo activado y actualizado el mapa de España (Mapsforge)
Sin embargo entro y veo que tengo un aviso "Mapa (2). Debo entender que tengo dos actualizaciones?

<img src="https://github.com/devemux86/cruiser/assets/122309655/57dc71cd-511c-4334-b25d-69efffcbea15" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/122309655/7e5f35eb-fe26-4b84-99f8-378f35d23b72" width="50%"/>

Si entro veo que en el icono de GraphHopper hay (1) aviso, a pesar de que no son mapas que tenga activados
<img src="https://github.com/devemux86/cruiser/assets/122309655/67e740d4-30f2-45c9-9372-e1809cd0527c" width="50%"/>

Si entro vuelvo a ver el mapa de España (Mapsforge) y si hago click en él, me pide activar o eliminar.
Aquí ya me he perdido.

<img src="https://github.com/devemux86/cruiser/assets/122309655/5c844c91-107b-40dd-8893-83a4d46dd951" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/122309655/4dbf6764-8463-47a7-90c0-349039bce38d" width="50%"/>

**Reply by devemux86** [reply]
@Cer-o 

Updates can be displayed for 3 elements: Maps, POI, GraphHopper.
In your 3rd image, GraphHopper has 1 update and probably the POI.
The POI update may be hidden due to the long name (we better rename this menu with a shorter name).
In your 4th image the updates appear on the 3rd screen with the clock icon.
Does this help you to understand all this?

Se pueden mostrar actualizaciones para 3 elementos: Mapas, POI, GraphHopper.
En tus terceras imágenes, GraphHopper tiene 1 actualización y probablemente el POI.
La actualización del POI puede estar oculta debido al largo nombre (mejor renombramos este menú con un nombre más corto).
En tu 4ª imagen las actualizaciones aparecen en la 3ª pantalla con el icono del reloj.
¿Te ayuda esto a entender todo esto?

**Comment by Cer-o** [comment]
They help me a lot, thank you.
And what exactly does GraphHopper do?

**Reply by devemux86** [reply]
> And what exactly does GraphHopper do?

Cruiser offers 2 offline routers: BRouter and GraphHopper.

It is recommended to use BRouter (offline).
It has more options and cross border routing.

Please see the BRouter guide for how to setup BRouter:
- https://github.com/devemux86/cruiser/discussions/137

**Comment by ToToGSARider** [comment]
Guten Morgen zusammen, ich wollte einfach mal Danke sagen @devemux86 für die Möglichkeit der offline Kartenaktualisierung. Es ist echt super, dass man jetzt einen Hinweis bekommt das neue Karten und POIs verfügbar sind. 
Morgen werde ich auch die erste lange Tour mit meiner offiziellen Cruiser App fahren. 

Good morning everyone, I just wanted to thank @devemux86 for the offline map update option. It's really great to get a notice now that new maps and POIs are available.
I will also be riding my first long route tomorrow with my official cruiser app.

**Reply by devemux86** [reply]
@ToToGSARider 

Thank you for the kind words and support! If you need anything, please let us know.

Vielen Dank für die freundlichen Worte und die Unterstützung! Wenn Sie etwas brauchen, lassen Sie es uns bitte wissen.
